changelogs/fragments/docfixes.yml

```diff
@@ -0,0 +1,2 @@
+bugfix:
+    - fixes ansible-doc, --json now is 'type intelligent' and reinstated httpapi, cliconf and --all options
```

Comment:
Reinstating httpapi and cliconf should be done in https://github.com/ansible/ansible/pull/47627